### PR TITLE
Movie details

### PR DESCRIPTION
**Pull request by FedericoChavazza**
## Links:

https://trello.com/c/FWRTgY7K/9-users-can-see-the-movie-details-by-clicking-on-a-poster

## What & Why:

Created view for everysingle movie detail, the data will be replaced once the api is given, I, too created a hook for horizontal scrolling if the images are more than 4, with a go back button, as asked in trello

image

## Risks, Mitigation & Rollback:

- [x] Safe to Revert *check this box if this PR can be reverted without incident*

### CODE REVIEW

**Review by JulianPasquale — COMMENTED** [review]
I see some repeated changes in other PRs. Which branch have you used to create this one?

To make reviewers life easier you should try to only keep the changes you are trying to add to the PR and avoid any non-related change so we don't have to review the exact same code more than once

File: src/components/common/Button/index.js [hunk 1]
```diff
@@ -1,12 +1,22 @@
...
-import './styles.css';
+import './styles.scss';
 
-const Button = ({ children, type = 'button', disabled, handleClick }) => (
-  <button className="button" onClick={handleClick} type={type} disabled={disabled}>
-    {children}
-  </button>
-);
+const Button = ({ img, children, type = 'button', disabled, handleClick }) => {
+  console.log(img);
```
Comment (JulianPasquale) [inline src/components/common/Button/index.js]: 🚫

File: src/hooks/useSideScroll.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { useRef, useEffect } from 'react';
+
+export function useHorizontalScroll() {
+  const elRef = useRef();
```
Comment (JulianPasquale) [inline src/hooks/useSideScroll.js]: Can we use more descriptive variables names?

File: src/pages/MovieInfo.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+            </div>
+          </div>
+        </div>
+      </div>
+      <div className="Movie-carousel__container">
+        <h2> {t('movieDetails.images')} </h2>
+        <div ref={scrollRef} className="MovieInfo__extraImgs">
+          {' '}
+          {mock[0]?.extraImgs.map((img, i) => {
+            console.log(img);
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: 🚫

File: src/services/endpoints/reduxEndpoints.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import endpoints from 'constants/endpoints';
```
Comment (JulianPasquale) [inline src/services/endpoints/reduxEndpoints.js]: Not needed for now, still using mock data for this task

File: src/state/slices/slice.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+import { createSlice } from '@reduxjs/toolkit';
```
Comment (JulianPasquale) [inline src/state/slices/slice.js]: Same here, remove this file

File: src/state/store/configureStore.dev.js [hunk 6]
```diff
@@ -2,7 +2,7 @@ import { configureStore } from '@reduxjs/toolkit';
 import logger from 'redux-logger';
 
 import reducer from 'state/reducer';
-import { api } from 'services/api';
+import { api } from 'services/endpoints/reduxEndpoints';
```
Comment (JulianPasquale) [inline src/state/store/configureStore.dev.js]: 🚫

File: src/state/reducer.js [hunk 7]
```diff
@@ -4,6 +4,7 @@ import { api } from 'services/api';
 
 const rootReducer = combineReducers({
   [api.reducerPath]: api.reducer,
+
```
Comment (JulianPasquale) [inline src/state/reducer.js]: 🚫

File: src/services/endpoints/reduxEndpoints.js [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+
+import { createApi } from '@reduxjs/toolkit/dist/query';
+
+export const api = createApi({
+  endpoints: builder => ({}),
+});
+
+// export const selectAuth = state => state.auth;
```
Comment (martoesses) [inline src/services/endpoints/reduxEndpoints.js]: check that there is no commented code when committing

File: src/services/endpoints/reduxEndpoints.js [hunk 9]
```diff
@@ -0,0 +1,9 @@
+import endpoints from 'constants/endpoints';
+
+import { createApi } from '@reduxjs/toolkit/dist/query';
```
Comment (martoesses) [inline src/services/endpoints/reduxEndpoints.js]: why you change '@reduxjs/toolkit/query/react' for '@reduxjs/toolkit/dist/query' again?

File: src/components/common/Button/index.js (hunk 1 above)
Comment (FedericoChavazza) [reply src/components/common/Button/index.js]: removed! sry

File: src/hooks/useSideScroll.js (hunk 2 above)
Comment (FedericoChavazza) [reply src/hooks/useSideScroll.js]: changed!

File: src/pages/MovieInfo.js (hunk 3 above)
Comment (FedericoChavazza) [reply src/pages/MovieInfo.js]: oops

File: src/App.js [hunk 10]
```diff
@@ -6,6 +6,8 @@ import useTranslation from 'hooks/useTranslation';
 
 import routes from 'routes';
 
+
```
Comment (JulianPasquale) [inline src/App.js]: Let's avoid this empty lines

File: src/hooks/useSideScroll.js [hunk 11]
```diff
@@ -0,0 +1,21 @@
+import { useRef, useEffect } from 'react';
+
+export function useHorizontalScroll() {
+  const elementRef = useRef();
+  useEffect(() => {
+    const el = elementRef.current;
```
Comment (JulianPasquale) [inline src/hooks/useSideScroll.js]: We still can improve vars naming here. Check the entire file

File: src/pages/MovieInfo.js [hunk 12]
```diff
@@ -0,0 +1,88 @@
...
+
+  const goBack = () => {
+    history.push(routesPaths.home);
+  };
+
+  const scrollRef = useHorizontalScroll();
+
+  return (
+    <div className="MovieInfo-container">
+      <Button handleClick={() => goBack()}>go back</Button>
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js]: Can we replace the "go back" text with I18n text?

File: src/pages/MovieInfo.js [hunk 13]
```diff
@@ -0,0 +1,88 @@
...
+import './styles.scss';
+import { AiOutlineStar } from 'react-icons/ai';
+import { BsFillBookmarkFill } from 'react-icons/bs';
+import { AiFillStar } from 'react-icons/ai';
+import { useHorizontalScroll } from 'hooks/useSideScroll';
+import useTranslation from 'hooks/useTranslation';
+import { useHistory } from 'react-router-dom';
+import routesPaths from 'routes/routesPaths';
+
+export function MovieInfo() {
```
Comment (JulianPasquale) [inline src/pages/MovieInfo.js:12]: Can we receive the movie here?
This way we won't need to remove `mock[0]` in the future (make the parent pass `mock[0]`)

File: src/pages/styles.scss [hunk 14]
```diff
@@ -0,0 +1,158 @@
...
+  flex-direction: column;
+  gap: 15px;
+}
+
+.MovieInfo__rate {
+  @include display_flex(center, flex-start);
+  background-color: #1d1d1d;
+  justify-content: center;
+  align-items: flex-start;
+  grid-gap: 30px;
```
Comment (JulianPasquale) [inline src/pages/styles.scss]: Should we also use `rem` here? cc: @martoesses

**Review by JulianPasquale — APPROVED** [review]
LGTM

Let's wait for @martoesses approval

**Review by martoesses — APPROVED** [review]

**Pull request merged**